core/src/main/java/tc/oc/pgm/namedecorations/NameDecorationRegistry.java

```diff
@@ -15,6 +16,17 @@
    */
   String getDecoratedName(Player player, Party party);
 
+  /**
+   * Get the fully decorated name as a Component
+   *
+   * <p>Note: Allows for prefix/suffix hover events
+   *
+   * @param player The player to decorate
+   * @param party The party this player is currently in
+   * @return The name, decorated, in component form
+   */
+  Component getDecoratedNameComponent(Player player, Party party);
```

Comment:
I'd do the same as above, provide a fallback default, even if the registry isn't really something that should be reimplemented by 3rd parties often